### PR TITLE
Simple fuzzer for nlohmann/json library

### PR DESCRIPTION
**Pull request by vitalybuka**

### CODE REVIEW

File: json/parse_fuzzer.cc [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+#include <json.hpp>
+
+using json = nlohmann::json;
+
+extern "C" int LLVMFuzzerTestOneInput(const uint8_t* data, size_t size) {
+  try {
+    std::stringstream s;
+    s << json::parse(data, data + size);
+    try {
+      auto j2 = json::parse(s.str());
```
Comment (mikea) [inline json/parse_fuzzer.cc]: are you trying to parse twice? why?

Comment (vitalybuka) [inline json/parse_fuzzer.cc]: To check that it can serialize parsed data, and found the bug https://github.com/nlohmann/json/issues/329

Comment (dvyukov) [inline json/parse_fuzzer.cc]: This technique finds tons of bugs and should be used for all libraries that has round-trip, i.e. serialization/deserialization, compression/decompression, conversion to/from, etc. Bugs found by such checks (logical bugs) are often valued by developers more than crashes on insane data.

I would also suggest to compare results of first and second parse. That catches losing, doubling, altering (e.g. double/missed escape/unescape) data during round-trip.

Comment (mikea) [inline json/parse_fuzzer.cc]: We should serialize j2 back then. S2 should be equal to s (but not data). Let's assert this as well.

Comment (vitalybuka) [inline json/parse_fuzzer.cc]: Done

**Pull request merged**